### PR TITLE
Do not throw exception but return nothing when there is no relationship with the given id

### PR DESCRIPTION
**Pull request by davidegrohmann**

### CODE REVIEW

File: community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v2_0/TransactionBoundQueryContext.scala [hunk 1]
```diff
@@ -191,7 +191,12 @@ class TransactionBoundQueryContext(graph: GraphDatabaseAPI, tx: Transaction, sta
       statement.dataWriteOperations().relationshipSetProperty(id, properties.Property.property(propertyKeyId, value) )
     }
 
-    def getById(id: Long) = graph.getRelationshipById(id)
+    def getById(id: Long) = try {
+      graph.getRelationshipById(id)
+    } catch {
+      case e: NotFoundException => throw new EntityNotFoundException(s"Relationship with id $id", e)
+      case e: RuntimeException  => throw e
```
Comment (pontusmelke) [inline community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v2_0/TransactionBoundQueryContext.scala]: What's the point of catching and rethrowing `RuntimeException`?

File: community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v2_1/TransactionBoundQueryContext.scala [hunk 2]
```diff
@@ -200,7 +200,12 @@ final class TransactionBoundQueryContext(graph: GraphDatabaseAPI,
       statement.dataWriteOperations().relationshipSetProperty(id, properties.Property.property(propertyKeyId, value) )
     }
 
-    def getById(id: Long) = graph.getRelationshipById(id)
+    def getById(id: Long) = try {
+      graph.getRelationshipById(id)
+    } catch {
+      case e: NotFoundException => throw new EntityNotFoundException(s"Relationship with id $id", e)
+      case e: RuntimeException  => throw e
```
Comment (pontusmelke) [inline community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v2_1/TransactionBoundQueryContext.scala]: Same here?

File: community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v2_2/TransactionBoundQueryContext.scala [hunk 3]
```diff
@@ -217,7 +217,12 @@ final class TransactionBoundQueryContext(graph: GraphDatabaseAPI,
       statement.dataWriteOperations().relationshipSetProperty(id, properties.Property.property(propertyKeyId, value) )
     }
 
-    def getById(id: Long) = graph.getRelationshipById(id)
+    def getById(id: Long) = try {
+      graph.getRelationshipById(id)
+    } catch {
+      case e: NotFoundException => throw new EntityNotFoundException(s"Relationship with id $id", e)
+      case e: RuntimeException  => throw e
```
Comment (pontusmelke) [inline community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v2_2/TransactionBoundQueryContext.scala]: …and here?

File: community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v2_0/TransactionBoundQueryContext.scala (hunk 1 above)
Comment (davidegrohmann) [inline community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v2_0/TransactionBoundQueryContext.scala]: Not sure actually, I have copied the code from the node `getById`.
I can clean it up in both places and see what happens.

**Pull request merged**